### PR TITLE
DOC: Typesetting of math for np.correlate and np.convolve

### PR DESCRIPTION
**Pull request by klieret**
This PR improves the documentation for

* `np.correlate`: Use math environment instead of formula shown as code
* `np.convolve`: Use more natural subscripts instead of brackets to denote item in sequence

I've checked that it displays correctly in the html.

### CODE REVIEW

**Comment by klieret** [comment]
The lint complains about escaping, but it's not necessary in math environments ([see here](https://github.com/sphinx-doc/sphinx/issues/2239#issuecomment-172341776)), so this is a false positive?

File: numpy/core/numeric.py [hunk 1]
```diff
@@ -679,12 +679,12 @@ def correlate(a, v, mode='valid'):
     Cross-correlation of two 1-dimensional sequences.
 
     This function computes the correlation as generally defined in signal
-    processing texts::
+    processing texts:
 
-        c_{av}[k] = sum_n a[n+k] * conj(v[n])
+    .. math:: c_k = \sum_n a_{n+k} \cdot \overline{v_n}
```
Comment (eric-wieser) [inline numpy/core/numeric.py:684]: You need to change the `"""` to `r"""` a few lines above, to stop python consuming the \

Comment (klieret) [reply numpy/core/numeric.py:684]: Ah, right of course! Thanks!

File: numpy/core/numeric.py [hunk 2]
```diff
@@ -804,7 +804,7 @@ def convolve(a, v, mode='full'):
     -----
     The discrete convolution operation is defined as
 
-    .. math:: (a * v)[n] = \\sum_{m = -\\infty}^{\\infty} a[m] v[n - m]
+    .. math:: (a * v)_n = \\sum_{m = -\\infty}^{\\infty} a_m v_{n - m}
```
Comment (mattip) [inline numpy/core/numeric.py:807]: Maybe make this consistent with the formula above?
```suggestion
    .. math:: c'_n = \\sum_{m = -\\infty}^{\\infty} a_m v_{n - m}
```

**Comment by klieret** [comment]
@eric-wieser The last failing check is already observed in https://github.com/numpy/numpy/commit/13e418073c567632a7ab9cff05fb767f784211c3, though I don't quite understand what causes it...

**Comment by charris** [comment]
The failing test looks unrelated. The error seems to be
```
/usr/bin/arm-linux-gnueabihf-gcc-ar: Cannot find plugin 'liblto_plugin.so'
```

**Review by eric-wieser — COMMENTED** [review]
There's another bit of misformatted TeX on line 741.

File: numpy/core/numeric.py (hunk 2 above)
Comment (klieret) [reply numpy/core/numeric.py:807]: Actually `np.convolve` doesn't introduce any `c`. Also the notation with `*` is pretty standard in mathematical texts, so I think it's clearer to keep it this way. What do you think?

**Comment by klieret** [comment]
> There's another bit of misformatted TeX on line 741.

Yes, this is a bit more tricky (because the `c` doesn't carry the `v` and `a` as indices).
I'd suggest just write this as plain python:

```python3
np.correlate(a, v) == np.conj(np.flip(np.correlate(v, a)))
```

or delete it (as it is already described and also obvious from the example).

**Comment by klieret** [comment]
> There's another bit of misformatted TeX on line 741.

I think this is fine now :)

**Comment by klieret** [comment]
Hi @mattip , @eric-wieser  I think we can merge this now :)

**Comment by mattip** [comment]
I find that the javascript rendering in the [new page](https://output.circle-artifacts.com/output/job/76281184-4a0c-466f-afc9-167e1c6f1476/artifacts/0/doc/build/html/reference/generated/numpy.correlate.html?highlight=correlate#numpy.correlate) is noticeably slow on both chromium and firefox on ubuntu 20.04. Maybe due to my slow internet. Is there something we can do to improve that? When I look at the developer tools, I see the MathJax javascript pulling in dozens of small javascript pieces, is there a bundle to pull them all in at once?

**Comment by klieret** [comment]
Thanks for checking that!

I believe the slow speed is due to the slow serving from circle-artifacts.com: For example you could compare the [stable docs on cov](https://numpy.org/doc/stable/reference/generated/numpy.cov.html?highlight=covariance) with the [circle docs on cov](https://output.circle-artifacts.com/output/job/76281184-4a0c-466f-afc9-167e1c6f1476/artifacts/0/doc/build/html/reference/generated/numpy.cov.html?highlight=covariance)  (which also makes use of LaTeX in the documentation).

It is indeed curious that so many different resources are being loaded, but I think this happens asynchronously, so most of the page should be there before math rendering is complete.

**Comment by klieret** [comment]
But now that I also compared it to other pages that use MathJax, I agree that the performance on the scipy documentation overall seems to be subpar. Perhaps because most other pages use the MathJax CDN?

Do you want to open a separate issue for that?

**Comment by mattip** [comment]
> Do you want to open a separate issue for that?

Sure

**Comment by InessaPawson** [comment]
Hi-five on merging your first pull request to NumPy, @klieret! We hope you stick around! Your choices aren’t limited to programming – you can review pull requests, help us stay on top of new and old  issues, develop educational material, work on our website, add or improve graphic design, create marketing materials, translate website content, write grant proposals, and help with other fundraising initiatives. For more info, check out: https://numpy.org/contribute
Also, consider joining our mailing list. This is a great way to connect with other cool people in our community and be part of important conversations that affect the development of NumPy: https://mail.python.org/mailman/listinfo/numpy-discussion

**Pull request merged**